src/components/PassphraseSetterBox.js

```diff
@@ -107,8 +117,10 @@ class PassphraseSetterBox extends Nimiq.Observable {
      */
     _onSubmit(event) {
         event.preventDefault();
-
-        if (!this._password) {
+        if (this._passphraseInput.text.length < PassphraseInput.DEFAULT_MIN_LENGTH) {
+            this.onPassphraseTooShort();
+            this.reset();
+        } else if (!this._password) {
```

Comment:
Could we just not allow submitting with enter if the password is too short? Just `return`, that's it? Because the button isn't there, so it shouldn't submit, just ignore the submitting.

You could then remove the `onPassphraseTooShort()` function.

Reply:
For passphrases < 8 it now shakes the hint and colors it red, without resetting the passphrase.
For repeating the passphrase it compares it with the first input as before.